packages/block-library/src/search/editor.scss

```diff
@@ -7,31 +7,16 @@
 		padding: $grid-unit-05;
 	}
 
-	.wp-block-search__input,
-	&.wp-block-search__button-inside .wp-block-search__inside-wrapper {
-		border-radius: $radius-block-ui;
-		border: 1px solid $gray-600;
-		color: $dark-gray-placeholder;
-		font-family: $default-font;
-		font-size: $default-font-size;
-		background-color: $white;
+	.wp-block-search__button {
+		height: auto;
+		border-radius: initial;
 
-		&:focus {
-			outline: none;
+		// This needs high specificity because it otherwise inherits styles from `components-button`.
+		&.wp-block-search__button.wp-block-search__button {
```

Comment:
@aristath I had to add --no-verify to be allowed to push this, per rules introduced in https://github.com/WordPress/gutenberg/pull/25399. I'm a big fan of linter rules, and the no duplicate selectors rule seems super fine. However the above also feels like a legitimate use case, doesn't it? That is — to add the extra specificity only for the one rule that needs it.

Is there some line I can write before this to allow it to pass the tests?

Reply:
> However the above also feels like a legitimate use case, doesn't it? That is — to add the extra specificity only for the one rule that needs it.

I agree. In fact I'd probably call this one a false-positive... The purpose of the rule is to avoid having multiple instances of the same selector in a file instead of combining them. In this case the double-selector is simply used to increase specificity :+1: 

> Is there some line I can write before this to allow it to pass the tests?

If I'm not mistaken, appending this to the line should ignore this failure: `// stylelint-disable-line no-duplicate-selectors`

Reply:
Let me try that one, thank you!

Reply:
Works! (⊃｡•́‿•̀｡)⊃━☆ﾟ.*･｡ﾟ

Reply:
It's a trick!

The padding value is overridden by `.components-button`, and while the ideal solution would be to refactor the block to not use that button at all (this is not UI, it's a block, after all), I deemed that out of scope for this PR.

But...

```
.wp-block-search__button {
}
```

has insufficient specificity. Whereas:

```
.wp-block-search__button.wp-block-search__button.wp-block-search__button {
}
```

this gets its specificity increased. It's a neat little trick to increase the specificity only to what you need. You can try it on https://specificity.keegan.st/.